shared/chat/conversation/messages/reactions-row/index.js

```diff
@@ -1,36 +1,81 @@
 // @flow
 import * as React from 'react'
 import * as Types from '../../../../constants/types/chat2'
-import {Box2} from '../../../../common-adapters'
-import ReactButton from '../react-button/with-tooltip'
-import {globalMargins, styleSheetCreate} from '../../../../styles'
+import {Box, Box2} from '../../../../common-adapters'
+import ReactButton from '../react-button/container'
+import ReactionTooltip from '../reaction-tooltip/container'
+import {collapseStyles, globalMargins, styleSheetCreate} from '../../../../styles'
 
 export type Props = {|
   conversationIDKey: Types.ConversationIDKey,
   emojis: Array<string>,
   ordinal: Types.Ordinal,
 |}
+type State = {
+  activeEmoji: string,
+  showAddReaction: boolean,
+}
+class ReactionsRow extends React.Component<Props, State> {
+  state = {
+    activeEmoji: '',
+    showAddReaction: false,
+  }
+  _attachmentRefs: {[emojiName: string]: ?React.Component<any, any>} = {}
 
-const ReactionsRow = (props: Props) =>
-  props.emojis.length === 0 ? null : (
-    <Box2 direction="horizontal" gap="xtiny" fullWidth={true} style={styles.container}>
-      {props.emojis.map(emoji => (
+  _setHoveringButton = (hovering: boolean, emojiName: string) => {
+    this._setActiveEmoji(hovering ? emojiName : '')
+  }
+
+  _setActiveEmoji = (emojiName: string) =>
+    this.setState(s => (s.activeEmoji === emojiName ? null : {activeEmoji: emojiName}))
+
+  _setHoveringRow = (hovering: boolean) =>
+    this.setState(s => (s.showAddReaction === hovering ? null : {showAddReaction: hovering}))
+
+  render() {
+    return this.props.emojis.length === 0 ? null : (
+      <Box2
+        onMouseOver={() => this._setHoveringRow(true)}
+        onMouseLeave={() => this._setHoveringRow(false)}
+        direction="horizontal"
+        gap="xtiny"
+        fullWidth={true}
+        style={styles.container}
+      >
+        {this.props.emojis.map(emoji => (
+          <Box
+            onMouseOver={() => this._setHoveringButton(true, emoji)}
+            onMouseLeave={() => this._setHoveringButton(false, emoji)}
+            key={emoji}
+          >
+            <ReactButton
+              ref={r => (this._attachmentRefs[emoji] = r)}
+              conversationIDKey={this.props.conversationIDKey}
+              emoji={emoji}
+              ordinal={this.props.ordinal}
+              style={styles.button}
+            />
+            {/* $FlowIssue */}
+            <ReactionTooltip
+              attachmentRef={this._attachmentRefs[emoji]}
+              conversationIDKey={this.props.conversationIDKey}
+              emoji={emoji}
+              onHidden={() => {}}
```

Comment:
we should just allow this to be undefined

Reply:
We need this prop for mobile, sorry should've `TODO`d here.